src/bundle/Command/MigrateNamespacesCommand.php

```diff
@@ -0,0 +1,206 @@
+        $this->output->writeln("Skipped $this->convertSkipped field attributes(s) which already had correct namespaces");
+    }
+
+    protected function getNextCursor(array $contentAttributeIDs): ?int
+    {
+        $lastId = count($contentAttributeIDs) > 0 ? end($contentAttributeIDs)['id'] : null;
+
+        return $lastId;
+    }
+
+    protected function processData($cursor)
+    {
+        $this->updateNamespacesInColumns($cursor['start'], $cursor['stop']);
+        if ($this->isChildProcess()) {
+            $this->output->writeln("Converted:$this->convertDone");
+            $this->output->writeln("Skipped:$this->convertSkipped");
+        }
+    }
+
+    protected function constructCursorFromInputOptions(): mixed
+    {
+        return [
+            'start' => $this->cursorStart,
+            'stop' => $this->cursorStop,
+        ];
+    }
+
+    protected function addChildProcessArguments($cursor): array
+    {
+        return [
+            '--cursor-start=' . $cursor['start'],
+            '--cursor-stop=' . $cursor['stop'],
+        ];
+    }
+
+    protected function isChildProcess(): bool
+    {
+        return $this->cursorStart !== null || $this->cursorStop !== null;
+    }
+
+    protected function processIncrementalOutput(string $output): void
+    {
+        if ($output !== '') {
+            $lines = explode(PHP_EOL, $output);
+            foreach ($lines as $line) {
+                if (strpos($line, 'Converted:') === 0) {
+                    $this->convertDone += (int) substr($line, strpos($line, ':') + 1);
+                } elseif (strpos($line, 'Skipped:') === 0) {
+                    $this->convertSkipped += (int) substr($line, strpos($line, ':') + 1);
+                } elseif ($line !== '') {
+                    $this->output->writeln($line);
+                }
+            }
+        }
+    }
+
+    protected function processIncrementalErrorOutput(string $output): void
+    {
+        $this->output->write($output);
+    }
+
+    public static function migrateNamespaces(string $xmlText)
+    {
+        $xmlText = str_replace('xmlns:ezxhtml="http://ez.no/xmlns/ezpublish/docbook/xhtml"', 'xmlns:ezxhtml="http://ibexa.co/xmlns/dxp/docbook/xhtml"', $xmlText);
+        $xmlText = str_replace('xmlns:ezcustom="http://ez.no/xmlns/ezpublish/docbook/custom"', 'xmlns:ezcustom="http://ibexa.co/xmlns/dxp/docbook/custom"', $xmlText);
+        $xmlText = str_replace('ezxhtml:class="ez-embed-type-image"', 'ezxhtml:class="ibexa-embed-type-image"', $xmlText);
+        $xmlText = str_replace('xmlns:ez="http://ez.no/xmlns/ezpublish/docbook"', 'xmlns:ez="http://ibexa.co/xmlns/ezpublish/docbook"', $xmlText);
+        $xmlText = str_replace('xmlns:a="http://ez.no/xmlns/annotation"', 'xmlns:a="http://ibexa.co/xmlns/annotation"', $xmlText);
+        $xmlText = str_replace('xmlns:m="http://ez.no/xmlns/module"', 'xmlns:m="http://ibexa.co/xmlns/module"', $xmlText);
```

Comment:
So does this mean that the entire essence of the changes is simple string replacement? Why don't we then execute it on the database directly? This should be 100x faster than pulling data into PHP process and update with new version. There are string replacement functions built-in DBMS-es.

Reply:
Correct place for the sql script would be https://github.com/ibexa/installer/tree/main/upgrade/db, right ?